crates/iroha/Cargo.toml

```diff
@@ -97,3 +97,4 @@ hex = { workspace = true }
 assertables = { workspace = true }
 trybuild = { workspace = true }
 assert_matches = "1.5.0"
+which = "7.0.3"
```

Comment:
nit: Consider also using `which 7.0.3` in [`iroha_test_network`](https://github.com/hyperledger-iroha/iroha/blob/5f8fd11c45a75e21174505f348932eb3d22f4320/crates/iroha_test_network/Cargo.toml#L38)

Reply:
Added to workspace deps instead.